api/v0/ingest/schema/types_test.go

```diff
@@ -0,0 +1,156 @@
+package schema_test
+
+import (
+	"math/rand"
+	"strings"
+	"testing"
+
+	stischema "github.com/filecoin-project/storetheindex/api/v0/ingest/schema"
+	"github.com/filecoin-project/storetheindex/test/util"
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-ipld-prime"
+	_ "github.com/ipld/go-ipld-prime/codec/dagjson"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/ipld/go-ipld-prime/node/bindnode"
+	"github.com/ipld/go-ipld-prime/storage/memstore"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_OldSchemaIsCompatibleWithNewStructs(t *testing.T) {
+	oldSchema, err := ipld.LoadSchemaBytes([]byte(`
+type EntryChunk struct {
+	Entries List_Bytes
+	Next optional Link_EntryChunk
+}
+
+type Link_EntryChunk &EntryChunk
+type List_String [String]
+type List_Bytes [Bytes]
+
+type Advertisement struct {
+	PreviousID optional Link_Advertisement
+	Provider String
+	Addresses List_String
+	Signature Bytes
+	Entries Link
+	ContextID Bytes
+	Metadata Bytes
+	IsRm Bool
+}
+
+type Link_Advertisement &Advertisement
+`))
```

Comment:
nit: I would add two levels of indentation to the schema string, which makes it easier to read the code, and still works as the schema DSL doesn't care about whitespace like that.